### ISSUE TITLE
Auto-Indent Loop termination

### ISSUE THREAD

**Issue by ajclarke90** [dparkins/language-fortran#30]
There seems to be an issue with loop indentation. In emacs:

```fortran
do i = 1, Ntot
    ...
end do
````

In atom using this package:

```fortran
do i = 1, Ntot
    ...
    end do 
````

This seems to be at least for if and do loops. Not sure if it effects anything else.

**Comment by tomedunn** [comment]
I just tested this on my machine using the current version of the language (v1.4.0) and the latest version of Atom (0.187.0) and it appears to be working properly for non-labeled do loops like the ones you provided. Are these the version of Atom/language-fortran you are using? For labeled do loops the indentation rules should be adjusted, at least for loops ending in an `end do` statement.
![image](https://cloud.githubusercontent.com/assets/5925666/6783839/0c110e28-d14a-11e4-9d24-5ac1975aedd4.png)
Are you running into this problem with labeled do loops, such as
```fortran
do 100, i = 1, 10
  ...
100 end do
```
or non-labeled do loops, such as
```fortran
do i = 1, 10
  ...
end do
```

**Comment by ajclarke90** [comment]
Non-labeled loops. I find the same behaviour as you if I test this in a new file, which probably means even though my code compiles somewhere, I must have either a broken syntax in a loop, or something else thats confusing the auto-indent feature.

**Comment by tomedunn** [comment]
Atom's auto indent feature looks at each line independently of the rest of the document. So in principle it shouldn't depend on what else you have in your code. If you run into the issue again try to check the current scope by pressing `ctrl+alt+shift+P` on windows or `cmd+shift+P` on mac. It'll pop up a little blue window like this
![image](https://cloud.githubusercontent.com/assets/5925666/6784396/6a0d34c2-d14d-11e4-953d-cdd89b056030.png)
Or if you have a file that produces the error, that you don't mind sharing, feel free to share it here.

**Comment by ajclarke90** [comment]
I'm getting sensible scopes:

![screen shot 2015-03-24 at 09 23 08](https://cloud.githubusercontent.com/assets/4687383/6798674/7e86c80e-d207-11e4-88bd-6504bbdea543.png)

Heres the code that it doesn;t work on:


```fortran
program ManyBriggs

  implicit none

  integer :: i, j, nw
  character*50 :: fn
  real :: Rd(3,3), Cnt( 3 )
  integer :: PGBEG, IER

  IER = PGBEG(0,'briggs.ps/cps',5,2)
  IF (IER.NE.1) STOP
  call PGPAP(0.0, 2./5.)

  print*,'How many do you want to plot?'
  read*,nw
  do j = 1, nw
    read*,fn
    call ReadPartsNative2(fn)
    call Centroid(Rd)
    do i = 1, 3
      Cnt( i ) = Rd(2,i)  !dark centroid
      end do
      call ShiftOriginVorX(Cnt,.false.,.false.)
      call Centroid(Rd)
      do i = 1, 3
        Cnt( i ) = Rd(2,i)  !dark centroid
        end do
        call RotIntoJz
        call BriggsFig2
        end do
        call PGEND
        end program ManyBriggs
```

**Comment by tomedunn** [comment]
I typed your code into Atom and it indented correctly. So I don't know why you seem to be getting different results. We can leave this issue open for the time being in case anyone else happens to run into the same problem.

**Comment by maxlevesque** [comment]
Hello world, I copy-pasted the code in Atom, and it does not indent correctly. I have the same uncorrect indentation as @ajclarke90 .

**Comment by tomedunn** [comment]
@maxlevesque if you highlight everything then go to `Edit > Lines > Auto Indent` does it fix it?

**Comment by maxlevesque** [comment]
That's exactly what I did. Sorry I was unclear in my previous post.
=> Auto Indent produces exactly the same badly indented file as @ajclarke90

**Comment by tomedunn** [comment]
Well I've tried it on both my windows machine and my mac and both work as expected. I'm guessing this is more of an Atom issue than it is a problem with language-fortran.

**Comment by ajclarke90** [comment]
Some further information in case it ever ends up helping someone.

If I type each line in individually, then atom aligns the do and end do perfectly and automatically:

````fortran
program test
  
  implicit none
  
  integer :: i 
  
  do i = 1,10
    print*, i
  end do
  
end program test
````

If I then go to this line and hit tab ( which for me is bound to auto-indent [ emacs-like behaviour]) then it indents to match the interior of the do loop:

```fortran
program test
  
  implicit none
  
  integer :: i 
  
  do i = 1,10
    print*, i
    end do
  
end program test
````

**Comment by tomedunn** [comment]
@ajclarke90, if you hit `ctrl+q` in Atom it will bring up the key binding resolver window. You can use it to check if you are actually auto-indenting when you are hitting tab.

**Comment by ajclarke90** [comment]
![screen shot 2015-03-26 at 12 53 41](https://cloud.githubusercontent.com/assets/4687383/6846588/3751f81c-d3b7-11e4-8ea1-875726412ebc.png)

I am. Its in my custom binds.
